pkg/fieldowner/field_owner.go

```diff
@@ -0,0 +1,46 @@
+package fieldowner
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Wrap wraps a Client with one that sets the field owner.
+func Wrap(upstream client.Client, owner string) client.Client {
+	return &clientWithOwner{Client: upstream, owner: client.FieldOwner(owner)}
+}
+
+type clientWithOwner struct {
+	client.Client
+	owner client.FieldOwner
+}
+
+func (c *clientWithOwner) Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+	return c.Client.Create(ctx, obj, append(opts, c.owner)...)
+}
+
+func (c *clientWithOwner) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	return c.Client.Update(ctx, obj, append(opts, c.owner)...)
+}
+
+func (c *clientWithOwner) Patch(ctx context.Context, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
```

Comment:
`Delete` doesn't need it? I intuitively would have expected `Delete` to make you owner of `metadata.DeletionTimestamp` (but haven't actually checked)

Reply:
Yeah, me too, but the [`FieldOwner`](https://github.com/kubernetes-sigs/controller-runtime/blob/7ef2da0bc161d823f084ad21ff5f9c9bd6b0cc39/pkg/client/options.go#L97-L113) type does not coerce to a [`DeleteOption`](https://github.com/kubernetes-sigs/controller-runtime/blob/7ef2da0bc161d823f084ad21ff5f9c9bd6b0cc39/pkg/client/options.go#L35).

I wonder if @DirectXMan12 who seems to have authored that in #435 has a view?